src/app/clusters/unit-localization/unit-localization.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/util/attribute-metadata.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+using namespace chip;
+using namespace chip::app::Clusters::UnitLocalization;
+
+using chip::app::ConcreteAttributePath;
+using Status = Protocols::InteractionModel::Status;
+
+Status MatterUnitLocalizationClusterServerPreAttributeChangedCallback(const ConcreteAttributePath & attributePath,
```

Comment:
Isn't this supposed to return an `EmberAfStatus`?

Reply:
Hmm, it looks like everyone does this, and I think this is only working because `ToInteractionModelStatus(EmberAfStatus code)` is just blindly casting it over...

Sigh.